src/js/components/LayerPanel/LayerRadio.js

```diff
@@ -24,11 +24,39 @@ export default class LayerRadio extends Component {
 
     this.layer = brApp.map.getLayer(props.layers[0].id);
     LayerActions.removeAllSubLayers.defer(this.layer);
+
     this.state = {
-      selected: this.layer.visibleLayers[0] || -1
+      selected: this.layer.visibleLayers[0] || -1,
+      exclusiveLayerIds: []
     };
   }
 
+  componentDidMount() {
+    // Get the list of unique layer ids that we need to turn off when a radio button is toggled on
+    const { settings } = this.context;
+    const { layerPanel } = settings;
+
+    const exclusiveLayerIds = settings.exclusiveRadioGroups
+    .reduce((result, currentVal) => {
+
+      return [
+        ...result,
+        ...layerPanel[currentVal].layers.map(l => l.id)
+        ];
+    }, [])
+    .filter(id => id !== this.layer.id);
+
+    const uniqueIds = [];
+    exclusiveLayerIds.forEach(id => {
+      if (uniqueIds.indexOf(id) === -1) uniqueIds.push(id);
```

Comment:
In what scenario would there be duplicates? Can track these or log them out somehow?